mne/decoding/search_light.py

```diff
@@ -144,7 +159,7 @@ def _transform(self, X, method):
             )
 
         y_pred = np.concatenate(y_pred, axis=1)
-        return y_pred
+        return y_pred.astype(X.dtype)
```

Comment:
this I would object. The predict dtype for a supervised approach should make the dtype of the y passed in fit. So if you do classification y is naturally int.

Reply:
Ahh I can make it more specific -- the error was this:
```
$ pytest mne/decoding/tests/test_search_light.py -k compli
...
mne/decoding/tests/test_search_light.py:339: in test_sklearn_compliance
    check(est)
../virtualenvs/base/lib/python3.11/site-packages/sklearn/utils/estimator_checks.py:1970: in check_transformer_preserve_dtypes
    assert Xt.dtype == dtype, (
E   AssertionError: SlidingEstimator (method=fit_transform) does not preserve dtype. Original/Expected dtype=float64, got dtype=int64.
```
so really I just need to move this to the `def transform(...)` function rather than the `_transform` function.

Reply:
... but I realize now that maybe you were already talking about `transform`. While I agree `int` is more natural unfortunately it looks like `sklearn` has made a different decision

Reply:
At least this way `.predict` is still `int` and `transform` is `float`, which I think is reasonable/correct...